library/platform.c

```diff
@@ -115,7 +115,7 @@ int mbedtls_platform_set_snprintf( int (*snprintf_func)( char * s, size_t n,
 }
 #endif /* MBEDTLS_PLATFORM_SNPRINTF_ALT */
 
-#if _WIN32_WINNT < 0x0A00 /* < Windows 10 */
+#if _WIN32 && _MSC_VER < 1900 /* Before MSVC 14.0 / Visual Studio 2015 */
```

Comment:
Why is the test on line 73 different from this one?

Reply:
The reason is the change being outside of this PR's scope, however it has been discussed that the changes could be more general.

Unfortunately I've noticed that there is MSYS mentioned as the reason for just testing for _WIN32. Which makes me want to make sure we have got the right filter here, as it may turn out the precise test whether the custom implementation is necessary or not happens to be complicated beyond feasibility.